### PR TITLE
implement auto login & make ProviderNotFoundException to be RuntimeException

### PR DESCRIPTION
**Pull request by vedi**
@refaelos, waiting for your code review

### CODE REVIEW

File: SoomlaAndroidProfile/src/com/soomla/profile/AuthController.java [hunk 1]
```diff
@@ -46,17 +47,19 @@
...
+    private static final String DB_KEY_PREFIX = "soomla.profile.common";
+
     /**
      * Constructor
      *
      * Loads all authentication providers
      * @param usingExternalProvider {@link SoomlaProfile#initialize}
      */
-    public AuthController(boolean usingExternalProvider, Map<IProvider.Provider, ? extends Map<String, String>> providerParams) {
+    public AuthController(boolean usingExternalProvider, Map<Object, Object> profileParams) {
```
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/AuthController.java]: Why did you have to change to <Object, Object>?

File: SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java [hunk 2]
```diff
@@ -45,34 +44,44 @@
...
 
+    private boolean autoLogin;
+
     /**
-     * see {@link #initialize(boolean, java.util.Map)}
+     * see {@link #initialize(Activity, boolean, Map)}
      */
-    public void initialize() {
-        initialize(null);
+    public void initialize(Activity activity) {
```
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java]: We're breaking backward compatibility here! Is it absolutely necessary?

File: SoomlaAndroidProfile/src/com/soomla/profile/exceptions/ProviderNotFoundException.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
  * This exception is thrown when the supplied provider is not supported
  * by the framework
  */
-public class ProviderNotFoundException extends Exception {
+public class ProviderNotFoundException extends RuntimeException {
```
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/exceptions/ProviderNotFoundException.java]: Why change it to RuntimeException and keep the "throws ProviderNotFoundException" everywhere in SoomlaProfile? It doesn't make much sense to me...

File: SoomlaAndroidProfile/src/com/soomla/profile/AuthController.java (hunk 1 above)
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/AuthController.java]: Ok I get why, but wouldn't it be better to actually just set the "autologin" param for each provider inside each of their params map? Who knows, we might want to change it sometime to be different for each provider

**Comment by vedi** [comment]
@borisspektor, @refaelos,
applied code review comments

**Comment by vedi** [comment]
@borisspektor, @refaelos, any feedback?

**Pull request merged**